libs/core/src/lib/calendar/calendar.component.ts

```diff
@@ -232,6 +256,58 @@ export class CalendarComponent<D> implements OnInit, ControlValueAccessor, Valid
     @Input()
     disableFunction: (date: D) => boolean = () => false;
 
+    /**
+     * Function used to disable previous button in the calendar header.
+     * @param date selected date
+     * @param currentlyDisplayedDate currently displayed date
+     * @param activeView current view of calendar
+     */
+    @Input()
+    get previousButtonDisableFunction(): NavigationButtonDisableFunction<D> {
+        return this._previousButtonDisableFunction;
+    }
+
+    set previousButtonDisableFunction(disableFunc: NavigationButtonDisableFunction<D>) {
+        if (typeof disableFunc === 'function') {
+            this._previousButtonDisableFunction = disableFunc;
+            if (this.currentlyDisplayed) {
+                this.previousButtonDisabled = this.previousButtonDisableFunction(
+                    this.selectedDate,
+                    this.currentlyDisplayed,
+                    this.activeView
+                );
+            }
+        } else {
+            throw new Error('previousButtonDisableFunction must be a function');
+        }
+    }
+
+    /**
+     * Function used to disable next button in the calendar header.
+     * @param date selected date
+     * @param currentlyDisplayedDate currently displayed date
+     * @param activeView current view of calendar
+     */
+    @Input()
+    get nextButtonDisableFunction(): NavigationButtonDisableFunction<D> {
+        return this._nextButtonDisableFunction;
+    }
+
+    set nextButtonDisableFunction(disableFunc: NavigationButtonDisableFunction<D>) {
+        if (typeof disableFunc === 'function') {
+            this._nextButtonDisableFunction = disableFunc;
+            if (this.currentlyDisplayed) {
+                this.nextButtonDisabled = this.nextButtonDisableFunction(
+                    this.selectedDate,
+                    this.currentlyDisplayed,
+                    this.activeView
+                );
+            }
```

Comment:
just wondering, wouldn't it be more concise to have just check for changed inputs inside `onChanges`?

It can look like this:
```typescript
ngOnChanges(changes: SimpleChanes): void {
  if ('nextButtonDisableFunction' in changes || 'previousButtonDisableFunction' in changes || 'currentlyDisplayed' in changes) {
      this. _setNavigationButtonsStates();
  }
}
```